Errors

davidvgalbraith opened the pull request:
Add an errors/ directory with fluxy actions etc. that is used to
render errors.

Completes https://github.com/juttle/outrigger/pull/69.

@mnibecker @go-oleg

mnibecker commented:
A couple of observations right off the bat
1. (Perhaps I am wrong) This doesn't seem to handle the case where there's a compilation error with my program.
2. Can you update `example/index.js` to use your error object?

davidvgalbraith commented:
If there's a compilation error we get here: https://github.com/juttle/juttle-client-library/pull/19/files#diff-5300733318314abc7552f68adb4dd9b7R55.

mnibecker commented:
Cool, I think what I was seeing is that if you call `input.render` on a program, stuff goes crazy.

mnibecker commented on src/errors/index.js:
```diff
@@ -0,0 +1,38 @@
+    render(err) {
+        let { dispatch } = this.store;
+        var info = err.info;
+        var error = (info && info.err && info.err.message) || err.message;
```
it would be nice to print out a title using the `code` attribute that outrigger api returns in all errors. I think this would help users distinguish between a compilation error, some outrigger api error, or some random js error that we caused.

mnibecker replied:
also you should use `let` instead of `var`

go-oleg commented on src/errors/error-view.js:
```diff
@@ -0,0 +1,24 @@
+import React, { Component } from "react";
+import { connect } from "react-redux";
+
+class ErrorView extends Component {
```
This is prolly something we could do elsewhere too, but when theres no `state` involved and only `props`, you can use a [stateless functional component](https://facebook.github.io/react/blog/2015/10/07/react-v0.14.html#stateless-functional-components). Less code ftw!

go-oleg commented on src/inputs/index.js:
```diff
@@ -36,6 +36,12 @@ export default class Input {
         this.store.dispatch(ActionCreators.updateBundle(bundle));
         return this.api.getInputs(bundle)
         .then(inputs => {
+            if (!Array.isArray(inputs)) {
```
Our `this.api.xxx` methods really should throw errors/reject when the response comes back with an error, but thats outside this PR.

davidvgalbraith commented:
OK @mnibecker I made inputs throw an appropriate error, updated the example and added a title and location info. And @go-oleg I used your strange magic for the ErrorView.

mnibecker commented:
looks good to merge.

Can you update the `CHANGELOG.md` and merge this into the `2.0` branch instead?
